SWITCHYARD-3012 - cloning the Fuse tooling Help files as a start

bfitzpat opened the pull request.

apupier commented on eclipse/plugins/org.switchyard.tools.help/.project:
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>org.switchyard.tools.help</name>
+	<comment>This is a shared POM parent for FuseSource Maven projects. NO_M2ECLIPSE_SUPPORT: Project files created with the maven-eclipse-plugin are not supported in M2Eclipse.</comment>
```
maybe FuseSource shoudl be removed?

apupier commented on eclipse/plugins/org.switchyard.tools.help/OSGI-INF/l10n/bundle.properties:
```diff
@@ -0,0 +1,3 @@
+#Properties file for org.fusesource.ide.help
+Bundle-Vendor = JBoss by Red Hat
+Bundle-Name = Fuse Plugins for Eclipse Help Plugin
```
SwitchYard Plugins?

apupier commented on eclipse/plugins/org.switchyard.tools.help/OSGI-INF/l10n/bundle.properties:
```diff
@@ -0,0 +1,3 @@
+#Properties file for org.fusesource.ide.help
```
org.fusesource.ide.help --> org.switchyard.tools.help

apupier commented on targetplatform/build.txt:
```diff
@@ -0,0 +1 @@
+[INFO] Scanning for projects...
```
I guess this file should not be commited

apupier commented on eclipse/plugins/org.switchyard.tools.help/pom.xml:
```diff
@@ -0,0 +1,42 @@
+               <target>
+                 <!-- copy the generated plugin -->
+                 <copy todir="${basedir}" failonerror="false" overwrite="true">
+                   <fileset dir="${basedir}/../../rider-docs/output/eclipsehelp" excludes="META-INF/**" />
```
is it something still used? there is no more "rider" folders as far as I know

lhein replied:
that looks outdated. if thats something we have in fuseide too we should get rid of it

bfitzpat replied:
any suggestions on what we should do with that? maybe just change to /output/eclipsehelp ?

lhein replied:
when thinking about it that could be something done by the docs team. We should ask Fintan and/or Jane.

bfitzpat replied:
Sent an e-mail to the docs team with the question.

fbolton replied:
I'm not too sure about this  either, as this POM is not part of our docs workflow. All I know is that 'rider' was an alternative brand for Apache Camel back in the FuseSource days (>5 years ago). So any instance of it sounds old to me.

bfitzpat replied:
@fbolton Any suggestions on what to do with this? Do we remove it? Do we still need the antrun?

fbolton replied:
My best guess is that this whole antrun thing is obsolete and can be removed. Just delete it and keep your fingers crossed!

apupier commented on eclipse/plugins/org.switchyard.tools.help/index.html:
```diff
@@ -0,0 +1,45 @@
+<html>
+   <head>
+      <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
+         <title>Red Hat JBoss Swithyard Tooling for Eclipse</title>
```
typo Swithyard
and shouldn't it be writtent a cap for Yard? SwitchYard?

apupier commented on eclipse/plugins/org.switchyard.tools.help/index.html:
```diff
@@ -0,0 +1,45 @@
+         <link rel="stylesheet" type="text/css" href="eclipse_book.css">
+            <meta name="generator" content="DocBook XSL Stylesheets V1.77.1">
+            <meta name="keywords" content="ESB, Open Source, open source, integration, OSGi">
+         <link rel="home" href="index.html" title="Red Hat JBoss Switchyard Tooling for Eclipse">
```
shouldn't it be writtent a cap for Yard? SwitchYard?

apupier commented on eclipse/plugins/org.switchyard.tools.help/index.html:
```diff
@@ -0,0 +1,45 @@
+            <div>
+               <div>
+                  <h1 class="title"><a name="SYEclipseHelp"></a>
+                     Red Hat JBoss Switchyard Tooling for Eclipse
```
shouldn't it be writtent a cap for Yard? SwitchYard?

apupier commented:
I think that the build failure is due to the fact that the plugins/pom.xml has not declared the new plugin module org.switchyard.tools.helpl

lhein commented on eclipse/features/org.switchyard.tools.feature/feature.xml:
```diff
@@ -4,9 +4,9 @@
       label="%featureName"
       version="2.4.0.qualifier"
       provider-name="%featureProvider"
+      plugin="org.switchyard.tools"
```
do u have a plugin with that id?

bfitzpat replied:
Yup

bfitzpat commented:
Testing-wise it works great locally in the Help system, but thanks for pointing out these issues. I'll take a crack at them.

bfitzpat commented:
Addressed all comments (thanks for the feedback and pointing out some of those issues in hidden files I missed) and sent an e-mail to the Docs team about the pom question.

lhein commented:
Please consult Jane. I know that docs team used to have some ant stuff
running. Just to be on the safe side.

Am 26.06.2017 6:24 nachm. schrieb "Fintan Bolton" <notifications@github.com
>:

> *@fbolton* commented on this pull request.
> ------------------------------
>
> In eclipse/plugins/org.switchyard.tools.help/pom.xml
> <https://github.com/jboss-switchyard/tools/pull/673#discussion_r124055708>
> :
>
> > +  <packaging>eclipse-plugin</packaging>
> +  <name>JBoss SwitchYard Tooling ::  Plugins :: Help</name>
> +
> +   <build>
> +     <plugins>
> +       <plugin>
> +       	 <groupId>org.apache.maven.plugins</groupId>
> +         <artifactId>maven-antrun-plugin</artifactId>
> +         <executions>
> +           <execution>
> +             <phase>package</phase>
> +             <configuration>
> +               <target>
> +                 <!-- copy the generated plugin -->
> +                 <copy todir="${basedir}" failonerror="false" overwrite="true">
> +                   <fileset dir="${basedir}/../../rider-docs/output/eclipsehelp" excludes="META-INF/**" />
>
> My best guess is that this whole antrun thing is obsolete and can be
> removed. Just delete it and keep your fingers crossed!
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jboss-switchyard/tools/pull/673#discussion_r124055708>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAamjeYVn5o3-st-ICl7T3i43mzTn5Frks5sH9s6gaJpZM4OClhu>
> .
>

bfitzpat commented:
Went ahead and removed the antrun build bits of the pom.xml for now. Everything seems to build ok locally. We can add them back in if we need to after hearing from Jane

bfitzpat commented:
According to Jane, we should listen to Fintan as the "master of doc Ant build tools." :) So hopefully this will build and then we'll get it merged after an approval from Lars or Aurelien. 👍
